src/com/tictacgo/data/Board.java

```diff
@@ -110,14 +112,11 @@ public Board(Board b) {
 		/**
 		 * Clones the pieces ArrayList
 		 */
-		pieces = new ArrayList<>(SIDE_LENGTH);
+		spaces = new ArrayList<>(SIDE_LENGTH);
 		for (int i = 0; i < SIDE_LENGTH; i++) {
```

Comment:
I don't think I can. There's no way to loop through b.spaces, and we haven't created this.spaces yet here.